From 0000000000000000000000000000000000000002 Mon Sep 17 00:00:00 2001
From: ravindu644
Date: Sun, 22 Feb 2026 22:24:27 +0530
Subject: [PATCH] xt_qtaguid: avoid kernel panic on inactive interfaces

Do not call dev_get_stats() on possibly inactive/stale net_dev from
iface_stat_fmt_proc_show(). Always use zeroed stats for this path.

Signed-off-by: ravindu644
---
 net/netfilter/xt_qtaguid.c | 10 +++-------
 1 file changed, 3 insertions(+), 7 deletions(-)

diff --git a/net/netfilter/xt_qtaguid.c b/net/netfilter/xt_qtaguid.c
--- a/net/netfilter/xt_qtaguid.c
+++ b/net/netfilter/xt_qtaguid.c
@@ -739,7 +739,7 @@ static int iface_stat_fmt_proc_show(struct seq_file *m, void *v)
 {
 	struct proc_iface_stat_fmt_info *p = m->private;
 	struct iface_stat *iface_entry;
-	struct rtnl_link_stats64 dev_stats, *stats;
+	struct rtnl_link_stats64 *stats;
 	struct rtnl_link_stats64 no_dev_stats = {0};
 
 	CT_DEBUG("qtaguid:proc iface_stat_fmt pid=%u tgid=%u uid=%u\n",
@@ -748,11 +748,7 @@ static int iface_stat_fmt_proc_show(struct seq_file *m, void *v)
 	iface_entry = list_entry(v, struct iface_stat, list);
 
-	if (iface_entry->active) {
-		stats = dev_get_stats(iface_entry->net_dev,
-				     &dev_stats);
-	} else {
-		stats = &no_dev_stats;
-	}
+	stats = &no_dev_stats;
+
 	/*
 	 * If the meaning of the data changes, then update the fmtX
 	 * string.
-- 
2.43.0
